### PR TITLE
Add repository nice-to-haves

### PR DESCRIPTION
**Pull request by michplunkett**
## Describe your changes
Added the `CODEOWNERS` file and a pull request template.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<!---
+Please write your PR name in the present imperative tense. Examples of present imperative tense are:
+"Fix issue in the dispatcher where…", "Improve our handling of…", etc."
+
+For more information on Pull Requests, you can reference here:
+https://success.vanillaforums.com/kb/articles/228-using-pull-requests-to-contribute
+-->
+## Describe your changes
```
Comment (michplunkett) [inline .github/pull_request_template.md:8]: Feel free to edit this as you want, but it sets a good baseline for pull requests.

**Review by FedericoDM — DISMISSED** [review]

**Comment by michplunkett** [comment]
I'm waiting to get the full-list of primaries and secondaries. Once I have that, I'll add the names, and then merge.

File: .github/pull_request_template.md (hunk 1 above)
Comment (aaronhaefner) [reply .github/pull_request_template.md:8]: Thanks, added a checklist too.

**Review by aaronhaefner — APPROVED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+## Non-obvious technical information
+
+
+## Checklist before requesting a review
+- [x] pre-commit run --all-files (run before pushing)
+- [x] pytest if applicable
+- [x] Link issue
+- [x] Update relevant documentation if applicable: doc strings, readme, poetry.
```
Comment (michplunkett) [inline .github/pull_request_template.md:18]: @aaronhaefner, for the template we'll want to make sure they are unchecked. Otherwise, LGTM!

**Review by aaronhaefner — APPROVED** [review]
Thank you for adding these!

**Review by FedericoDM — APPROVED** [review]

**Review by klinkoberstar — APPROVED** [review]

**Pull request merged**